source/custom/nidaqmx_service.custom.cpp

```diff
@@ -1,4 +1,12 @@
 #include <nidaqmx/nidaqmx_service.h>
 
 namespace nidaqmx_grpc {
+
+::grpc::Status NiDAQmxService::ConvertApiErrorStatusForTaskHandle(google::protobuf::int32 status, TaskHandle task)
+{
+    std::string description(nidevice_grpc::kMaxGrpcErrorDescriptionSize, '\0');
+    library_->GetErrorString(status, description.data(), nidevice_grpc::kMaxGrpcErrorDescriptionSize);
```

Comment:
`DAQmxGetErrorString` returns a static error message. For dynamic errors, call `DAQmxGetExtendedErrorInfo`.

Reply:
Thanks, I guess as long as it's on the same thread, that will always correctly return the last error for that thread?

Reply:
Yes, `DAQmxGetExtendedErrorInfo` returns the last error for the current thread.